### PR TITLE
tests: New unit tests for submoduling

### PR DESCRIPTION
**Pull request by zpan1**
New tests for: Array, ClassificationMetric, CorrelationTest, KNN, NaiveBayes, RandomForest, SVM
Also some fixes

### CODE REVIEW

**Comment by devksingh4** [comment]
@zpan1 and potentially @ltcptgeneral: why is github actions not running the unit tests?

**Comment by ltcptgeneral** [comment]
Gonna wait to review until floating point issues are resolved

File: analysis-master/test_analysis.py [hunk 1]
```diff
@@ -30,6 +48,60 @@ def test_():
...
+	assert classif_metric[1] == metrics.classification_report(test_data_linear, test_data_linear2)
+	"""
+	# there will be floating point precision issues with the tests below
+	assert CorrelationTest.anova_oneway(test_data_linear, test_data_linear2) == {"f-value": 0.05825242718446602, "p-value": 0.8153507906592907}
+	assert CorrelationTest.pearson(test_data_linear, test_data_linear2) == {'r-value': 0.9153061540753287, 'p-value': 0.02920895440940868}
+	assert CorrelationTest.spearman(test_data_linear, test_data_linear2) == {"r-value":0.9746794344808964, "p-value":0.004818230468198537}
+	assert CorrelationTest.point_biserial(test_data_linear, test_data_linear2) == {'r-value': 0.9153061540753287, 'p-value': 0.02920895440940868}
+	assert CorrelationTest.kendall(test_data_linear, test_data_linear2) == {"tau":0.9486832980505137, "p-value":0.022977401503206086}
+	assert CorrelationTest.kendall_weighted(test_data_linear, test_data_linear2) == {"tau":0.9750538072369643, "p-value":np.nan}
+	#assert CorrelationTest.mgc()
```
Comment (ltcptgeneral) [inline analysis-master/test_analysis.py:71]: fix floating point issues

**Review by ltcptgeneral — CHANGES_REQUESTED** [review]

**Review by ltcptgeneral — APPROVED** [review]
LGTM.

**Comment by ltcptgeneral** [comment]
close #51

**Pull request merged**